create-regression-sprint-report/RegressionSprintReport/index.ts

```diff
@@ -0,0 +1,264 @@
+        'Content-Type': 'application/json'
+    }
+    try {
+        const response = await fetch(confluenceUrl, {
+            method: 'POST',
+            body: body,
+            headers: headers
+        });
+
+        return response;
+    
+    } catch (error) {
+        console.log(error)
+    }
+}
+
+function createPageContent(modulesList: ModuleInfo[]){
+
+    console.log("Creating Confluence page content");
+
+    let pageBody: string = '<table><tbody>';
+    let rowN: number = 0;
+
+    pageBody += '<tr> <th>#</th> <th>Module Name</th> <th>Module Version</th> <th>Type testing</th> <th>Issues from previous sprints</th> <th>Issues</th> </tr>'
+
+    for(let module of modulesList){
+        pageBody += `<tr> <td>${++ rowN}</td> <td>${module.id}</td> <td>${module.version}</td> <td></td> <td>${module.oldIssues}</td> <td>${module.sprintIssues}</td> </tr>`
+    }
+    pageBody += '</tbody></table>';
+
+    return pageBody;
+}
+
+function createPageSettings(pageType: string, pageTitle: string, spaceKey: string, subpageId: number, pageBody: string, pageRepresentation: string): ConfluencePageInfo{
+
+    const pageSettings: ConfluencePageInfo = 
+    {
+        type: pageType,
+        title: pageTitle,
+        ancestors:[
+            {
+                id: subpageId
+            }
+        ], 
+        space:{
+            key: spaceKey
+        },
+        body:{
+            storage:{
+                value: pageBody,
+                representation: pageRepresentation
+            }
+        }
+    };
+    return pageSettings;
+}
+
+const httpTrigger: AzureFunction = async function (context: Context, req: HttpRequest): Promise<void> {
+
+    context.log('HTTP trigger function processed a request.');
+    
+    const login = process.env[loginSecretParam];
+    const password = process.env[passwordSecretParam];
+    const confluenceUrl = process.env[confluenceUrlParam];
+    const jiraUrl = process.env[jiraUrlParam];
+
+    const configMapUrl = req.body && req.body.configMapUrl;
+    const kustomizationUrl = req.body && req.body.kustomizationUrl;
+    const pageType = req.body && req.body.pageType; 
+    const pageTitle = req.body && req.body.pageTitle; 
+    const spaceKey = req.body && req.body.spaceKey; 
+    const subpageId = req.body && req.body.subpageId; 
+    const pageRepresentation = req.body && req.body.pageRepresentation;
+
+    const issueType = req.body && req.body.issueType;
+    const parentIssueKey = req.body && req.body.parentIssueKey;
+
+    let modulesList: any[] = configMapUrl ? await getModulesList(configMapUrl) : null;
+    modulesList = modulesList ? modulesList.sort((a,b) => (a.id > b.id) ? 1 : ((b.id > a.id) ? -1 : 0)): null;
+    if (modulesList) {
+        modulesList.unshift(await getPlatformVersion(kustomizationUrl));
+    }
+    modulesList = modulesList ? await fillModulesIssues(modulesList, parentIssueKey, issueType, jiraUrl, login, password) : null;

```

Comment:
Try to remove all nulls and redundant conditions. Check for nulls inside functions and implement guard clause https://deviq.com/design-patterns/guard-clause.

Reply:
fixed